fmriprep/workflows/bold/base.py

```diff
@@ -0,0 +1,785 @@
+            ('subject_id', 'inputnode.subject_id'),
+            ('t1_2_fsnative_reverse_transform', 'inputnode.t1_2_fsnative_reverse_transform')
+        ]),
+        (inputnode, bold_confounds_wf, [('t1_tpms', 'inputnode.t1_tpms'),
+                                        ('t1_mask', 'inputnode.t1_mask')]),
+        (bold_hmc_wf, bold_reg_wf, [('outputnode.bold_split', 'inputnode.bold_split'),
+                                    ('outputnode.xforms', 'inputnode.hmc_xforms')]),
+        (bold_hmc_wf, bold_confounds_wf, [('outputnode.movpar_file', 'inputnode.movpar_file')]),
+        (bold_reg_wf, bold_confounds_wf, [
+            ('outputnode.bold_t1', 'inputnode.bold_t1'),
+            ('outputnode.bold_mask_t1', 'inputnode.bold_mask_t1')]),
+        (bold_reference_wf, func_reports_wf, [
+            ('outputnode.validation_report', 'inputnode.validation_report')]),
+        (bold_reg_wf, func_reports_wf, [
+            ('outputnode.out_report', 'inputnode.bold_reg_report'),
+            ('outputnode.fallback', 'inputnode.bold_reg_fallback'),
+        ]),
+        (bold_confounds_wf, outputnode, [
+            ('outputnode.confounds_file', 'confounds'),
+            ('outputnode.aroma_noise_ics', 'aroma_noise_ics'),
+            ('outputnode.melodic_mix', 'melodic_mix'),
+            ('outputnode.nonaggr_denoised_file', 'nonaggr_denoised_file'),
+        ]),
+        (bold_reg_wf, outputnode, [('outputnode.bold_t1', 'bold_t1'),
+                                   ('outputnode.bold_mask_t1', 'bold_mask_t1')]),
+        (bold_confounds_wf, func_reports_wf, [
+            ('outputnode.acompcor_report', 'inputnode.acompcor_report'),
+            ('outputnode.tcompcor_report', 'inputnode.tcompcor_report'),
+            ('outputnode.ica_aroma_report', 'inputnode.ica_aroma_report')]),
+        (bold_confounds_wf, summary, [('outputnode.confounds_list', 'confounds')]),
+        (bold_reg_wf, summary, [('outputnode.fallback', 'fallback')]),
+        (summary, func_reports_wf, [('out_report', 'inputnode.summary_report')]),
+    ])
+
+    # bool('TooShort') == True, so check True explicitly
+    if run_stc is True:
+        workflow.connect([
+            (bold_reference_wf, bold_stc_wf, [('outputnode.bold_file', 'inputnode.bold_file'),
+                                              ('outputnode.skip_vols', 'inputnode.skip_vols')]),
+            (bold_stc_wf, bold_hmc_wf, [('outputnode.stc_file', 'inputnode.bold_file')])])
+    else:
+        workflow.connect([
+            (bold_reference_wf, bold_hmc_wf, [('outputnode.bold_file', 'inputnode.bold_file')])])
+
+    # Cases:
+    # fmaps | use_syn | force_syn  |  ACTION
+    # ----------------------------------------------
+    #   T   |    *    |     T      | Fieldmaps + SyN
+    #   T   |    *    |     F      | Fieldmaps
+    #   F   |    *    |     T      | SyN
+    #   F   |    T    |     F      | SyN
+    #   F   |    F    |     F      | HMC only
```

Comment:
If you're going to do this, should we make the choice of fieldmap unwarping and  SyN-based unwarping part of the fieldmaps workflows? Though, that may make more sense as a separate PR, as this one is already pretty big. Either way.

Reply:
Yes - the SyN-based workflow should be attached the same way we are currently doing with the fieldmap decision maker in `workflows.fieldmap.base` if we want to be consistent. But better in a different PR where that's the only target.